### ISSUE TITLE
rename function arguments

### ISSUE THREAD

**Issue by adokter** [adokter/bioRad#112]
So far we have focussed on renaming the function names, but the function argument names also need improvements. E.g. `integrate_profile` has `alt.min` argument, which still has the dot convention we want to deprecate. This is still quite a lot of work..., but I think we cannot push anything to master until this is done.

We could contact the editors of Ecograpy to forward install instructions to the editors of the paper to a branch that has not yet renamed arguments, only function names. Then we don't have to release to the public yet and have more time.

## Functions

* [ ] beam.R: `beam_height(range, elev, k = 4/3, lat = 35, re = 6378, rp = 6357)`
* [ ] beam.R: `beam_width(range, angle = 1)`
* [ ] bioRad.R: no functions
* [ ] bioRad-deprecated: arguments should remain the same
* [ ] bind_into_vpts.R: `bind_into_vpts(x, ...)`
* [ ] bind_into_vpts.R: `bind_into_vpts(...)`
* [ ] bind_into_vpts.R: `bind_into_vpts(x, ...)`
* [ ] bind_into_vpts.R: `bind_into_vpts(..., attributes_from = 1)`
* [ ] calculate_vp.R: `calculate_vp(vol.in, vp.out = "", vol.out = "", autoconf = FALSE, verbose = FALSE, mount = dirname(vol.in), sd_vvp = 2, rcs = 11, dualpol = FALSE, rhohv = 0.95, elev.min = 0, elev.max = 90, azim.min = 0, azim.max = 360, range.min = 5000, range.max = 25000, nlayer = 20L, hlayer = 200, dealias = TRUE, nyquist.min = if (dealias) 5 else 25)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] check_night.R: `check_night(lon, lat, date, elev = -0.268)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] composite_ppi.R: `composite_ppi(x, param = "DBZH", cells.dim = c(100, 100))`
* [ ] data.R: no functions
* [ ] dbz_eta.R: `dbz_to_eta(dbz, wavelength, Km = 0.93)`
* [ ] dbz_eta.R: `eta_to_dbz(eta, wavelength, Km = 0.93)`
* [ ] docker.R: `check_docker(verbose = TRUE)`
* [ ] docker.R: `update_docker()`
* [ ] download_basemap.R: `download_basemap(x, verbose = TRUE, zoom, alpha = 1, ...)`
* [ ] download_vpfiles.R: `download_vpfiles(start_date, end_date, country, radar, localpath = ".")`
* [ ] get_elevation_angles.R: `get_elevation_angles(x)`
* [ ] get_elevation_angles.R: `get_elevation_angles(x)`
* [ ] get_elevation_angles.R: `get_elevation_angles(x)`
* [ ] get_quantity.R: `get_quantity(x, quantity)`
* [ ] get_quantity.R: `get_quantity(x, quantity = "dens")`
* [ ] get_quantity.R: `get_quantity(x, quantity = "dens")`
* [ ] get_quantity.R: `get_quantity(x, quantity = "dens")`
* [ ] get_scan.R: `get_scan(x, angle)`
* [ ] integrate_profile.R: `integrate_profile(x, alt.min, alt.max, alpha = NA, interval.max = Inf)`
* [ ] integrate_profile.R: `integrate_profile(x, alt.min = 0, alt.max = Inf, alpha = NA, interval.max = Inf)`
* [ ] integrate_profile.R: `integrate_profile(x, alt.min = 0, alt.max = Inf, alpha = NA, interval.max = Inf)`
* [ ] integrate_profile.R: `integrate_profile(x, alt.min = 0, alt.max = Inf, alpha = NA, interval.max = Inf)`
* [ ] map.R: `map(x, ...)`
* [ ] map.R: `map(x, map, param, alpha = 0.7, xlim, ylim, zlim = c(-20, 20), ratio, radar.size = 3, radar.color = "red", n.color = 1000, ...)`
* [ ] nexrad_odim.R: `nexrad_to_odim(vol.in, vol.out, verbose = FALSE, mount = dirname(vol.in))`
* [ ] param.R: `summary(object, ...)`
* [ ] param.R: `is.param(x)`
* [ ] param.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] plot.ppi.R: `plot(x, param, xlim, ylim, zlim = c(-20, 20), ratio = 1, ...)`
* [ ] plot.scan.R: `plot(x, param, xlim = c(0, 100), ylim = c(0, 360), zlim = c(-20, 20), ...)`
* [ ] plot.vp.R: `plot(x, quantity = "dens", xlab = expression("volume density [#/km"^3 * "]"), ylab = "height [km]", line.col = "red", line.lwd = 1, ...)`
* [ ] plot.vpi.R: `plot(x, quantity = "mtr", xlab = "time", ylab = "migration traffic rate [#/km/h]", main = "MTR", nightshade = TRUE, elev = -0.268, lat = NULL, lon = NULL, ylim = NULL, ...)`
* [ ] plot.vpts.R: `plot(x, xlab = "time", ylab = "height [m]", quantity = "dens", log = T, barbs = T, barbs.h = 10, barbs.t = 20, barbs.dens = 5, zlim, legend.ticks, main, ...)`
* [ ] ppi.R: `[(x, i)`
* [ ] ppi.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] ppi.R: `summary(object, ...)`
* [ ] ppi.R: `is.ppi(x)`
* [ ] ppi.R: `dim(x)`
* [ ] project_as_ppi.R: `project_as_ppi(x, cellsize = 500, range.max = 50000, project = FALSE, latlim = NULL, lonlim = NULL)`
* [ ] project_as_ppi.R: `project_as_ppi(x, cellsize = 500, range.max = 50000, project = FALSE, latlim = NULL, lonlim = NULL)`
* [ ] project_as_ppi.R: `project_as_ppi(x, cellsize = 500, range.max = 50000, project = FALSE, latlim = NULL, lonlim = NULL)`
* [ ] pvol.R: `summary(object, ...)`
* [ ] pvol.R: `is.pvol(x)`
* [ ] pvol.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] pvolfile.R: `is.pvolfile(filename)`
* [ ] pvolfile.R: `get_odim_object_type(filename)`
* [ ] rcs.R: `rcs(x) <- value`
* [ ] rcs.R: `rcs(x)`
* [ ] read_pvolfile.R: `read_pvolfile(filename, param = c("DBZH", "VRADH", "VRAD", "RHOHV", "ZDR", "PHIDP", "CELL"), sort = TRUE, lat, lon, height, elangle.min = 0, elangle.max = 90, verbose = TRUE, mount = dirname(filename))`
* [ ] read_vpfiles.R: `read_vpfiles(files)`
* [ ] read_vpts.R: `read_vpts(file, radar, wavelength = "C")`
* [ ] regularize_vpts.R: `regularize_vpts(ts, interval = "auto", t.min = ts$daterange[1], t.max = ts$daterange[2], units = "mins", fill = FALSE, verbose = TRUE)`
* [ ] rvsd.R: `rvsd(x) <- value`
* [ ] rvsd.R: `rvsd(x)`
* [ ] scan.R: `summary(object, ...)`
* [ ] scan.R: `is.scan(x)`
* [ ] scan.R: `dim(x)`
* [ ] scan.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] select_vpfiles.R: `select_vpfiles(path, start_date, end_date, country = NULL, radar = NULL)`
* [ ] sunrise_sunset.R: `sunrise(lon, lat, date, elev = -0.268)`
* [ ] sunrise_sunset.R: `sunset(lon, lat, date, elev = -0.268)`
* [ ] traffic.R: all deprecated functions
* [ ] vp.R: `summary(object, ...)`
* [ ] vp.R: `is.vp(x)`
* [ ] vp.R: `dim(x)`
* [ ] vp.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] vp.R: `as.data.frame(x, row.names = NULL, optional = FALSE, quantities = names(x$data), suntime = TRUE, geo = TRUE, elev = -0.268, lat = NULL, lon = NULL, ...)`
* [ ] vp.R: `c(...)`
* [ ] vpfile.R: `is.vpfile(filename)`
* [ ] vpts.R: `summary(object, ...)`
* [ ] vpts.R: `is.vpts(x)`
* [ ] vpts.R: `dim(x)`
* [ ] vpts.R: `[(x, i)`
* [ ] vpts.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] vpts.R: `as.data.frame(x, row.names = NULL, optional = FALSE, quantities = names(x$data), suntime = TRUE, geo = TRUE, elev = -0.268, lat = NULL, lon = NULL, ...)`

## Internal functions

* [ ] color_scale.R: `color_scale(param, zlim)`
* [ ] color_scale.R: `color_scale_fill(param, zlim)`
* [ ] color_scale.R: `add_color_transparency(col, alpha = 1)`
* [ ] beam.R: `earth_radius(a, b, latdeg)`
* [ ] bind_into_vpts.R: `vplist_to_vpts(x, radar = NA)`
* [ ] bind_into_vpts.R: `vp_to_vpts_helper(vps)`
* [ ] docker.R: `mount_docker_container(mount = "~/")`
* [ ] download_vpfiles.R: `check_url_existence(url)`
* [ ] map.R: `get_zlim(param)`
* [ ] nexrad_odim.R: `nexrad_to_odim_tempfile(vol.in, verbose = FALSE, mount = dirname(vol.in))`
* [ ] project_as_ppi.R: `sample_polar(param, cellsize, range.max, project, latlim, lonlim)`
* [ ] project_as_ppi.R: `wgs_to_proj(lon, lat, proj4string)`
* [ ] project_as_ppi.R: `proj_to_wgs(x, y, proj4string)`
* [ ] project_as_ppi.R: `cartesian_to_polar(coords, elev = 0)`
* [ ] project_as_ppi.R: `safe_subset(data, indexx, indexy)`
* [ ] project_as_ppi.R: `polar_to_index(coords.polar, rangebin = 1, azimbin = 1)`
* [ ] pvolfile.R: `is.odimfile(filename)`
* [ ] read_pvolfile.R: `read_pvolfile_scan(filename, scan, param, radar, datetime, geo)`
* [ ] read_pvolfile.R: `read_pvolfile_quantity(filename, quantity, radar, datetime, geo)`
* [ ] read_vpfiles.R: `read_vp(filename)`: function should be hidden
* [ ] read_vpfiles.R: `quantity_name(file, group)`
* [ ] read_vpfiles.R: `read_odim_profile_data(file, group)`
* [ ] select_vpfiles: `match_filenames((filelist, regexlist))`
* [ ] sunrise_sunset.R: `get_suntime(lon, lat, date, sign_angle, elev = -0.268)`
* [ ] vpts.R: `vpts_to_vp(x, i)`

**Comment by adokter** [comment]
See https://github.com/adokter/bioRad/issues/51#issuecomment-374747768 for how to handle deprecation/renaming of arguments such that package remains backward compatible

**Comment by adokter** [comment]
This issue is related to issue #94. I noticed some functions have been ticked there, even whentheir parameter names are not optimal (e.g. containing dots)

**Comment by peterdesmet** [comment]
Agreed, it is something we should tackle: I’ll add a todo list to this issue.

Ok for creating branch and contacting editor. I’ll let you do that. Would be nice if #108 and #117 are accepted and included.

Updated #94: that should only concern documentation, not parameters.

**Comment by adokter** [comment]
#108 and #117 have been merged into develop

**Comment by peterdesmet** [comment]
@adokter, we'll let you make a suggestion for standardized and logical parameter names, before we provide further input.

**Comment by adokter** [comment]
## Renamed functions arguments

* [x] beam.R: `beam_height(range, elev, k = 4/3, lat = 35, re = 6378, rp = 6357)`
* [x] beam.R: `beam_width(range, beam_angle = 1)`
* [ ] bioRad.R: no functions
* [ ] bioRad-deprecated: arguments should remain the same
* [ ] bind_into_vpts.R: `bind_into_vpts(x, ...)`
* [ ] bind_into_vpts.R: `bind_into_vpts(...)`
* [ ] bind_into_vpts.R: `bind_into_vpts(x, ...)`
* [ ] bind_into_vpts.R: `bind_into_vpts(..., attributes_from = 1)`
* [x] calculate_vp.R: `calculate_vp(pvolfile, vpfile = "", pvolfile_out = "", autoconf = FALSE, verbose = FALSE, mount = dirname(pvolfile), sd_vvp = 2, rcs = 11, dual_pol = FALSE, rho_hv = 0.95, elev_min = 0, elev_max = 90, azim_min = 0, azim_max = 360, range_min = 5000, range_max = 25000, n_layer = 20L, h_layer = 200, dealias = TRUE, nyquist_min = if (dealias) 5 else 25)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] check_night.R: `check_night(lon, lat, date, elev = -0.268)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] check_night.R: `check_night(x, ..., elev = -0.268)`
* [ ] composite_ppi.R: `composite_ppi(x, quantity = "DBZH", dim = c(100, 100))`
* [ ] data.R: no functions
* [x] dbz_eta.R: `dbz_to_eta(dbz, wavelength, K = 0.93)`
* [x] dbz_eta.R: `eta_to_dbz(eta, wavelength, K = 0.93)`
* [ ] docker.R: `check_docker(verbose = TRUE)`
* [ ] docker.R: `update_docker()`
* [ ] download_basemap.R: `download_basemap(x, verbose = TRUE, zoom, alpha = 1, ...)`
* [x] download_vpfiles.R: `download_vpfiles(date_min, date_max, country, radar, localpath = ".")`
* [ ] get_elevation_angles.R: `get_elevation_angles(x)`
* [ ] get_elevation_angles.R: `get_elevation_angles(x)`
* [ ] get_elevation_angles.R: `get_elevation_angles(x)`
* [ ] get_quantity.R: `get_quantity(x, quantity)`
* [ ] get_quantity.R: `get_quantity(x, quantity = "dens")`
* [ ] get_quantity.R: `get_quantity(x, quantity = "dens")`
* [ ] get_quantity.R: `get_quantity(x, quantity = "dens")`
* [x] get_scan.R: `get_scan(x, elev)`
* [x] integrate_profile.R: `integrate_profile(x, alt_min, alt.max, alpha = NA, interval_max = Inf)`
* [x] integrate_profile.R: `integrate_profile(x, alt_min = 0, alt_max = Inf, alpha = NA, interval_max = Inf)`
* [x] integrate_profile.R: `integrate_profile(x, alt_min = 0, alt_max = Inf, alpha = NA, interval_max = Inf)`
* [x] integrate_profile.R: `integrate_profile(x, alt_min = 0, alt_max = Inf, alpha = NA, interval_max = Inf)`
* [ ] map.R: `map(x, ...)`
* [x] map.R: `map(x, map, quantity, alpha = 0.7, xlim, ylim, zlim = c(-20, 20), ratio, radar_size = 3, radar_color = "red", n_color = 1000, ...)`
* [x] nexrad_odim.R: `nexrad_to_odim(pvolfile_nexrad, pvolfile_odim, verbose = FALSE, mount = dirname(pvolfile_nexrad))`
* [ ] param.R: `summary(object, ...)`
* [ ] param.R: `is.param(x)`
* [ ] param.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [x] plot.ppi.R: `plot(x, quantity, xlim, ylim, zlim = c(-20, 20), ratio = 1, ...)`
* [x] plot.scan.R: `plot(x, quantity, xlim = c(0, 100), ylim = c(0, 360), zlim = c(-20, 20), ...)`
* [ ] plot.vp.R: `plot(x, quantity = "dens", xlab = expression("volume density [#/km"^3 * "]"), ylab = "height [km]", line_col = "red", line_lwd = 1, ...)`
* [x] plot.vpi.R: `plot(x, quantity = "mtr", xlab = "time", ylab = "migration traffic rate [#/km/h]", main = "MTR", night_shade = TRUE, elev = -0.268, lat = NULL, lon = NULL, ylim = NULL, ...)`
* [x] plot.vpts.R: `plot(x, xlab = "time", ylab = "height [m]", quantity = "dens", log = T, barbs = T, barbs_height = 10, barbs_time = 20, barbs_dens_min = 5, zlim, legend_ticks, main, ...)`
* [ ] ppi.R: `[(x, i)`
* [ ] ppi.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] ppi.R: `summary(object, ...)`
* [ ] ppi.R: `is.ppi(x)`
* [ ] ppi.R: `dim(x)`
* [x] project_as_ppi.R: `project_as_ppi(x, grid_size = 500, range_max = 50000, project = FALSE, ylim = NULL, xlim = NULL)`
* [x] project_as_ppi.R: `project_as_ppi(x, grid_size = 500, range_max = 50000, project = FALSE, ylim = NULL, xlim = NULL)`
* [x] project_as_ppi.R: `project_as_ppi(x, grid_size = 500, range_max = 50000, project = FALSE, ylim = NULL, xlim = NULL)`
* [ ] pvol.R: `summary(object, ...)`
* [ ] pvol.R: `is.pvol(x)`
* [ ] pvol.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] pvolfile.R: `is.pvolfile(filename)`
* [x] pvolfile.R: `get_odim_object_type(file)`
* [ ] rcs.R: `rcs(x) <- value`
* [ ] rcs.R: `rcs(x)`
* [x] read_pvolfile.R: `read_pvolfile(filename, param = c("DBZH", "VRADH", "VRAD", "RHOHV", "ZDR", "PHIDP", "CELL"), sort = TRUE, lat, lon, height, elev_min = 0, elev_max = 90, verbose = TRUE, mount = dirname(filename))`
* [ ] read_vpfiles.R: `read_vpfiles(files)`
* [ ] read_vpts.R: `read_vpts(file, radar, wavelength = "C")`
* [x] regularize_vpts.R: `regularize_vpts(ts, interval = "auto", date_min = ts$daterange[1], date_max = ts$daterange[2], units = "mins", fill = FALSE, verbose = TRUE)`
* [x] sd_vvp.R: `sd_vvp(x) <- value`
* [x] sd_vvp.R: `sd_vvp(x)`
* [ ] scan.R: `summary(object, ...)`
* [ ] scan.R: `is.scan(x)`
* [ ] scan.R: `dim(x)`
* [ ] scan.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [x] select_vpfiles.R: `select_vpfiles(path, date_min, date_max, country = NULL, radar = NULL)`
* [ ] sunrise_sunset.R: `sunrise(lon, lat, date, elev = -0.268)`
* [ ] sunrise_sunset.R: `sunset(lon, lat, date, elev = -0.268)`
* [ ] traffic.R: all deprecated functions
* [ ] vp.R: `summary(object, ...)`
* [ ] vp.R: `is.vp(x)`
* [ ] vp.R: `dim(x)`
* [ ] vp.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] vp.R: `as.data.frame(x, row.names = NULL, optional = FALSE, quantities = names(x$data), suntime = TRUE, geo = TRUE, elev = -0.268, lat = NULL, lon = NULL, ...)`
* [ ] vp.R: `c(...)`
* [ ] vpfile.R: `is.vpfile(filename)`
* [ ] vpts.R: `summary(object, ...)`
* [ ] vpts.R: `is.vpts(x)`
* [ ] vpts.R: `dim(x)`
* [ ] vpts.R: `[(x, i)`
* [ ] vpts.R: `print(x, digits = max(3L, getOption("digits") - 3L), ...)`
* [ ] vpts.R: `as.data.frame(x, row.names = NULL, optional = FALSE, quantities = names(x$data), suntime = TRUE, geo = TRUE, elev = -0.268, lat = NULL, lon = NULL, ...)`

## Internal functions

* [x] color_scale.R: `color_scale(quantity, zlim)`
* [x] color_scale.R: `color_scale_fill(quantity, zlim)`
* [ ] color_scale.R: `add_color_transparency(col, alpha = 1)`
* [ ] beam.R: `earth_radius(a, b, latdeg)`
* [ ] bind_into_vpts.R: `vplist_to_vpts, radar = NA)`
* [ ] bind_into_vpts.R: `vp_to_vpts_helper(vps)`
* [ ] docker.R: `mount_docker_container(mount = "~/")`
* [ ] download_vpfiles.R: `check_url_existence(url)`
* [x] map.R: `get_zlim(quantity)`
* [x] nexrad_odim.R: `nexrad_to_odim_tempfile(pvolfile, verbose = FALSE, mount = dirname(pvolfile))`
* [x] project_as_ppi.R: `sample_polar(quantity, grid_size, range_max, project, latlim, lonlim)`
* [ ] project_as_ppi.R: `wgs_to_proj(lon, lat, proj4string)`
* [ ] project_as_ppi.R: `proj_to_wgs(x, y, proj4string)`
* [ ] project_as_ppi.R: `cartesian_to_polar(coords, elev = 0)`
* [ ] project_as_ppi.R: `safe_subset(data, indexx, indexy)`
* [x] project_as_ppi.R: `polar_to_index(coords_polar, rangebin = 1, azimbin = 1)`
* [ ] pvolfile.R: `is.odimfile(filename)`
* [ ] read_pvolfile.R: `read_pvolfile_scan(filename, scan, param, radar, datetime, geo)`
* [ ] read_pvolfile.R: `read_pvolfile_quantity(filename, quantity, radar, datetime, geo)`
* [ ] read_vpfiles.R: `read_vp(filename)`: function should be hidden
* [ ] read_vpfiles.R: `quantity_name(file, group)`
* [ ] read_vpfiles.R: `read_odim_profile_data(file, group)`
* [ ] select_vpfiles: `match_filenames((filelist, regexlist))`
* [ ] sunrise_sunset.R: `get_suntime(lon, lat, date, sign_angle, elev = -0.268)`
* [ ] vpts.R: `vpts_to_vp(x, i)`

@peterdesmet first iteration of new argument names above. I checked the ones that I changed

**Comment by adokter** [comment]
@peterdesmet @stijnvanhoey Any chance you can have a look at this before the training school in Oklahoma starting August 14? I'd like to push for a new release by then with corrected function arguments.

**Comment by peterdesmet** [comment]
Hi @adokter I created a file with all the parameters for all functions: [parameters.txt](https://github.com/adokter/bioRad/files/2227134/parameters.txt). Makes it easier to compare these. Here's my review:

Be consistent with underscore:

- [x] all parameters with `min`/`max` use underscore 👍👍
- [ ] `azimbin` -> `azim_bin` (cf. `azim_min`/`azim_max`)
- [ ] `rangebin` -> `range_bin` (cf. `range_min`/`range_max`)

Use underscore where it might make sense (i.e. these parameters don't have an underscore yet, but it would makes sense to do it). These are suggestions:

- [ ] `row.names` -> `row_names`
- [ ] `indexx` -> `index_x`
- [ ] `indexy` -> `index_y`
- [ ] `latdeg` -> `lat_deg` (is there reason not to use `lat` here?)
- [ ] `latlim` -> `lat_lim`
- [ ] `lonlim` -> `lon_lim`
- [ ] `xlab` -> `x_lab` (what does the `lab` stand for?)
- [ ] `xlim` -> `x_lim`
- [ ] `ylab` -> `y_lab`
- [ ] `ylim` -> `y_lim`
- [ ] `zlim` -> `z_lim`

What is the difference between (should they get the same name?):

- [ ] `proj4string` vs `project`
- [ ] `file` vs `filename`
- [ ] `files` vs `filelist`
- [ ] `K` vs `k`
- [ ] `localpath` vs `path`
- [ ] `re` vs `regexlist`: no idea if these are related?

Unintuitive names:

- [ ] `sign_angle`: is this plus/minus?
- [ ] `autoconf`: is this auto configuration?
- [ ] `line_lwd`: lwd?
- [ ] `rho_hv`
- [ ] `rp`
- [ ] `ts`
- [ ] `vps`: vp objects?

Maybe write in full:

- [ ] `coords` -> `coordinates`
- [ ] `coords_polar` -> `coordinates_polar`

**Comment by peterdesmet** [comment]
Ok!

1.  Can you update the name suggestions that you approved (azim_bin, range_bin)
2. Keep `row.names`: ok
3. `xlab` and co: fine to keep as is. @stijnvanhoey is on holiday
4. `proj4string`: if part of proj4, maybe call it `proj4_string` then?
5. Let's use `file` then?
6. Let's use `files` then?
7. Let's use `path` or maybe `directory` (to differentiate from `file`) then?
8. `coords_polar`: would only update if we change `coords` as well, so best to keep it as is.

**Comment by adokter** [comment]
Sounds good these names, added comments in your comment above

**Comment by peterdesmet** [comment]
4. `proj4string`: ok to keep

5/6. I would keep the difference between `file` and `files`. Using `file` for all makes it less clear which functions _do_ support _more_ than a single file (and not all functions do). For those that do support more, I think `files` is intuitive enough that you can supply a list of one. Note regarding `filename` vs `file`: [readr](https://readr.tidyverse.org/reference/read_delim.html) also uses `file`, which aligns nicely. 👍  Combined with `directory`, we would get:

```r
download_vpfiles(start_date, end_date, country, radar, directory = ".")
is.pvolfile(file)
get_odim_object_type(file)
read_pvolfile(file, ...)
read_vpfiles(files)
read_vpts(file, ...)
read_vpfiles(files)
read_vpts(file, ...)
select_vpfiles(directory, ...)
is.vpfile(file)

# internal
is.odimfile(file)
read_pvolfile_scan(file, ...)
read_pvolfile_quantity(file, ...)
read_vp(file)
quantity_name(file, ...)
read_odim_profile_data(file, ...)
match_filenames((filelist, regexlist)) # this is an internal function, I would keep the arguments as such, maybe but with underscore
```

8. I would also update `col` to `color` in the internal function `add_color_transparency(col, alpha = 1)`

If you agree, can you update the list at https://github.com/adokter/bioRad/issues/112#issuecomment-394833402, so we have an overview of what needs to be changed?

**Comment by adokter** [comment]
Update the list at https://github.com/adokter/bioRad/issues/112#issuecomment-394833402. Only check items have changes

**Comment by stijnvanhoey** [comment]
Regarding 

```
read_pvolfile.R: read_pvolfile(file, param = c("DBZH", "VRADH", "VRAD", "RHOHV", "ZDR", "PHIDP", "CELL"), sort = TRUE, lat, lon, height, elev_min = 0, elev_max = 90, verbose = TRUE, mount = dirname(filename))
```

should this not be `quantity` instead of `param`?

**Comment by stijnvanhoey** [comment]
Regarding the `param` versus `quantity` usage as argument, I do think we are introducing some inconsistencies with the above proposed set:

```
param.R                                       # scan parameter
read_pvolfile.R: read_pvolfile(file, param,...)  # scan parameter
composite_ppi(x, quantity = "DBZH",...)  # scan parameter <- ISSUE
get_quantity(...)    # quantity of a vp/vpts
map.R: map(x, map, quantity,...)  # scan parameter  <- ISSUE
plot.ppi.R: plot(x, quantity,...) # scan parameter  <- ISSUE
plot.scan.R: plot(x, quantity,...) # scan parameter  <- ISSUE
plot.vp.R: plot(x, quantity = "dens",...) # quantity of a vp/vpts
plot.vpi.R: plot(x, quantity = "mtr",...)   # quantity 
plot.vpts.R: plot(x, xlab = "time", ylab = "height [m]", quantity,...)  # quantity 
```

For each of the items with `<- ISSUE` I will NOT use `quantity` as argument, but param, corresponding to the scan parameter.

**Comment by adokter** [comment]
Regarding `param` versus `quantity` in https://github.com/adokter/bioRad/issues/112#issuecomment-409850808, it might actually be more easy for users to always have `quantity`, the difference between a scan parameter or a derived quantity is pretty subtle, and most people might not understand it.

**Comment by adokter** [comment]
https://github.com/adokter/bioRad/issues/112#issuecomment-409827982: you're right this might be confusing. I would be ok with replace `latlim` with `ylim` and `lonlim` with `xlim`

**Comment by stijnvanhoey** [comment]
While implementing and checking docs etc. I felt having a difference between the scan parameters according to the OPERA data information versus vp/vpts/... quantities is less confusing than quantities everywhere (more intuition than real argument). @peterdesmet any thoughts?

**Comment by peterdesmet** [comment]
I think we have had the `param` vs `quantity` argument before and opted to keep `param` as that is the (unchangeable) name that is used in a pvol file: `pvol > scan > param`?

There might be some gain in using a single name for `param`/`quantity`, but not sure how much. I notice that in the [bird profile](https://github.com/adokter/vol2bird/wiki/ODIM-bird-profile-format-specification) things are called `quantity`. @adokter I'll let you decide this one.

**Comment by peterdesmet** [comment]
Note: I have updated https://github.com/adokter/bioRad/issues/112#issuecomment-394833402 to only include those functions that we change (you can still see the full list at the beginning of this issue or in the version history of the comment). I'll check them off when reviewing https://github.com/adokter/bioRad/pull/127.

**Comment by adokter** [comment]
@peterdesmet @stijnvanhoey I think we have to go with `quantity` only. Currently indeed we have `quantity` refer to different data columns in a `vp`, and `param` refer to different data sweeps in a `scan`. But these get mixed up already, e.g. we have `DBZH` as a quantity in the profile, but also as a param in a scan. Also, I might add in the future the option to plot scan parameters like DBZH as a transformed quantity (e.g. reflectivity eta, which is currently only a quantity in a profile).

In short: the two are getting mixed up already, so I think the least confusing is to stick to one.

**Comment by peterdesmet** [comment]
Makes sense, “quantity” it is. @stijnvanhoey, can you update this in #127? @adokter, does this affect the figure/text in the paper?

**Comment by stijnvanhoey** [comment]
So, if we only keep `quantity`, what happens with the [file `param.R`](https://github.com/adokter/bioRad/blob/argument-renaming/R/param.R)?

```
#' Object of class \code{param}: a parameter of a scan of a polar volume
#'
...
#' @details
#' Scan parameters are simple matrices, with the following specific attributes:
```

will this becoome a `quantity` class? and do we speak about `Scan quantities are...`?

I do think the class `param` is counterintuitive with using scan quantitites.

**Comment by peterdesmet** [comment]
@adokter https://github.com/adokter/bioRad/pull/127 was merged, but there is still the open issue of scan vs quantity to be resolved. Let us know what you decide.

**Comment by adokter** [comment]
Hi @peterdesmet, you're right that one is still open, @stijnvanhoey got me doubting again. Letting go of param breaks the OPERA/ODIM convention, which formed the backbone of the naming conventions. So I'm hesitant to introduce `scan quantities` when OPERA uses `scan parameters`. So maybe there is sense in keeping param strictly for the quantities in the low-level data after all (polar volumes and scans), as you wrote in https://github.com/adokter/bioRad/issues/112#issuecomment-411070067, and use quantity strictly for data columns in profiles, and for derived/transformed quantities.

I already have (soon to be included in bioRad) functionality that makes rasters (stored as ppi objects) of derived quantities (i.e. raster images of vertically integrated bird density, using information of multiple elevations scans to make one image). However, these ppi's always have only one quantity in the raster, therefore there is no need to specify the quantity when plotting them with `map` or `plot.ppi`, and no param or quantity argument is needed when plotting these types of ppi's. A ppi with multiple raster layers in fact only occurs now when making a ppi of a scan, and these only contain params. 

So I think we can keep param, and nothing needs to be changed.

Here a listing of functions with param/quantity arguements for future reference
* color_scale.R:`color_scale`: param (ok for now, has only param color scales)
* color_scale.R: `color_scale_fill`: param (ok for now, has only param color scales)
* composite_ppi.R: `composite_ppi`: param
* map.R: `map`: param
* map.R: `get_zlim`: param (ok for now, checks only param scales)
* plot.ppi.R: `plot.ppi`: param
* plot.scan.R: `plot.scan`: param
* plot.vp.R: `plot.vp`: quantity
* plot.vpi.R: `plot.vpi`: quantity
* project_as_ppi.R: `sample_polar`: param

**Comment by adokter** [comment]
think we can close this one:)

**Comment by peterdesmet** [comment]
Nice! I added a label `deprecation` to this issue for future reference. See https://github.com/adokter/bioRad/issues/112#issuecomment-411789178
